qstash/features/callbacks.mdx

```diff
@@ -82,3 +83,41 @@ your `Max Message Size` in the
 Make sure you verify the authenticity of the callback request made to your API
 by
 [verifying the signature](/qstash/features/security/#request-signing-optional).
+
+# What is a Failure-Callback?
+
+<Info>Failure Callbacks are only available in v2</Info>
+
+Failure callback is similar to callbacks but they are called only when all the retries are exhausted and still 
+the message can not be delivered to the given endpoint.
+
+This is designed to be an alternative to [List messages to DLQ](/qstash/api/dlq/listMessages). See [Migration Quide](/qstash/migrating/v2#where-are-the-messages-in-v2) for more details for the detailed flow how to use them.
+
+You can add a failure callback url in the `Upstash-Failure-Callback` header when publishing a
+message. The value must be a valid URL.
+
+```bash
+curl -X POST \
+  https://qstash.upstash.com/v2/publish/<DESTINATION_URL> \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <QSTASH_TOKEN>' \
+  -H 'Upstash-Failure-Callback: <CALLBACK_URL>' \
+  -d '{ "hello": "world" }'
+```
+
+The callback body sent to you will be a JSON object with the following fields:
+
+```json
+{
+  "status": 200,
+  "sourceMessasgeId": "msg_xxx", // The ID of the message that triggered the callback
+  "headers": {
```

Comment:
```suggestion
  "header": {
```

because the struct as defined below:
```go
type CallbackMessage struct {
	Status          int         `json:"status"`
	SourceMessageId string      `json:"sourceMessageId"`
	Retried         int32       `json:"retried,omitempty"`
	Header          http.Header `json:"header,omitempty"`
	Body            []byte      `json:"body,omitempty"`
	MaxRetries      int32       `json:"maxRetries"`
	DlqId           string      `json:"dlqId,omitempty"`
}
```